app/models/user.rb

```diff
@@ -16,7 +16,7 @@ class User < ApplicationRecord
   has_many   :miq_widget_sets, :as => :owner, :dependent => :destroy
   has_many   :miq_reports, :dependent => :nullify
   has_many   :service_orders, :dependent => :nullify
-  has_many   :shares
+  has_many   :owned_shares, :class_name => "Share"
```

Comment:
I believe that grep is the built-in Enumerable#grep, which uses `===` under the covers, which enables that class like lookup (as well as the more expected regex and other comparisons).